src/clients/HubPoolClient.ts

```diff
@@ -244,17 +254,32 @@ export class HubPoolClient extends BaseAbstractClient {
   }
 
   async computeRealizedLpFeePct(
-    deposit: { quoteTimestamp: number; amount: BigNumber; destinationChainId: number; originChainId: number },
+    deposit: {
+      quoteTimestamp: number;
+      amount: BigNumber;
+      destinationChainId: number;
+      originChainId: number;
+      blockNumber: number;
+    },
     l1Token: string
   ): Promise<{ realizedLpFeePct: BigNumber | undefined; quoteBlock: number }> {
     const quoteBlock = await this.getBlockNumber(deposit.quoteTimestamp);
     if (!quoteBlock) {
       throw new Error(`Could not find block for timestamp ${deposit.quoteTimestamp}`);
     }
 
-    const version = this.configStoreClient.getConfigStoreVersionForTimestamp(deposit.quoteTimestamp);
-    if (isUBA(version)) {
-      if (!this.configStoreClient.hasLatestConfigStoreVersion) {
+    // To determine if a deposit should be applied a UBA fee, we need to check the start block of the bundle
+    // that would contain this deposit.
+    const bundleStartBlockContainingDeposit = this.getBundleStartBlockContainingBlock(
+      deposit.blockNumber,
+      deposit.originChainId,
+      this.latestBlockNumber
+    );
+    const versionAppliedToDeposit = this.configStoreClient.getConfigStoreVersionForBlock(
+      bundleStartBlockContainingDeposit
+    );
+    if (isUBA(versionAppliedToDeposit)) {
+      if (!this.configStoreClient.isValidConfigStoreVersion(versionAppliedToDeposit)) {
```

Comment:
I see so we're essentially determining this on a per-flow basis? Good catch

Reply:
yea i think that is more correct I realized. If a deposit happened after the UBA activation block but is contained in a bundle who's start block is before the UBA activation block, then the deposit can't have a UBA fee applied to it because of the way we compute running balances